File: dropwizard/src/main/java/com/yammer/dropwizard/jersey/params/LongParam.java
package com.yammer.dropwizard.jersey.params;

// TODO: 11/14/11 <coda> -- document LongParam
// TODO: 11/14/11 <coda> -- test LongParam

public class LongParam extends AbstractParam<Long> {
    public LongParam(String input) {
        super(input);
    }

    @Override
    protected String errorMessage(String input, Exception e) {
        return '"' + input + "\" is not a number.";
    }

    @Override
    protected Long parse(String input) {
        return Long.valueOf(input);
    }
}

File: dropwizard-scala_2.9.1/src/main/scala/com/yammer/dropwizard/scala/params/LongParam.scala
package com.yammer.dropwizard.scala.params

object LongParam {
  def apply(value: Long): LongParam = LongParam(value.toString)
}

/**
 * Parses longs.
 */
case class LongParam(s: String) extends AbstractParam[Long](s) {
  protected def parse(input: String) = input.toLong

  override protected def renderError(input: String, e: Throwable) =
    "Invalid parameter: %s (Must be an integer value.)".format(input)
}
